$MirOS: ports/lang/python/2.5/patches/patch-Lib_test_test_fcntl_py,v 1.1 2008/07/15 19:18:53 bsiegert Exp $
$OpenBSD: patch-Lib_test_test_fcntl_py,v 1.1.1.1 2003/12/31 17:38:33 sturm Exp $
--- Lib/test/test_fcntl.py.orig	Tue Sep  5 02:54:19 2006
+++ Lib/test/test_fcntl.py	Wed May 28 20:48:35 2014
@@ -24,8 +24,7 @@ if sys.platform in ('netbsd1', 'netbsd2'
                     'Darwin1.2', 'darwin',
                     'freebsd2', 'freebsd3', 'freebsd4', 'freebsd5',
                     'freebsd6', 'freebsd7',
-                    'bsdos2', 'bsdos3', 'bsdos4',
-                    'openbsd', 'openbsd2', 'openbsd3', 'openbsd4'):
+                    'bsdos2', 'bsdos3', 'bsdos4'):
     if struct.calcsize('l') == 8:
         off_t = 'l'
         pid_t = 'i'
@@ -33,6 +32,18 @@ if sys.platform in ('netbsd1', 'netbsd2'
         off_t = 'lxxxx'
         pid_t = 'l'
     lockdata = struct.pack(off_t+off_t+pid_t+'hh', 0, 0, 0, fcntl.F_WRLCK, 0)
+elif sys.platform.startswith('mirbsd'):
+    if struct.calcsize('l') == 8:
+        off_t = 'l'
+    else:
+        off_t = 'lxxxx'
+    pid_t = 'i'
+    lockdata = struct.pack(off_t+off_t+pid_t+'hh', 0, 0, 0, fcntl.F_WRLCK, 0)
+elif sys.platform in ['openbsd', 'openbsd2', 'openbsd3']:
+    if sys.maxint == 2147483647:
+        lockdata = struct.pack('lxxxxlxxxxlhh', 0, 0, 0, fcntl.F_WRLCK, 0)
+    else:
+        lockdata = struct.pack('ixxxxixxxxihh', 0, 0, 0, fcntl.F_WRLCK, 0)
 elif sys.platform in ['aix3', 'aix4', 'hp-uxB', 'unixware7']:
     lockdata = struct.pack('hhlllii', fcntl.F_WRLCK, 0, 0, 0, 0, 0, 0)
 elif sys.platform in ['os2emx']:
